experimental/dds/tree2/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -721,7 +722,8 @@ export class ModularChangeFamily
 			return new Map();
 		}
 
-		return this.intoDeltaImpl(change.fieldChanges);
+		const idAllocator = MemoizedIdRangeAllocator.fromNextId() as unknown as MemoizedIdAllocator;
```

Comment:
It seems pretty unfortunate to have to cast to unknown when `MemoizedIdRangeAllocator` is only being used for this purpose. I'm not convinced it's worth decoupling the allocator from the branded types since this is a relatively specialized utility.

Reply:
Yeah, I wanted to avoid the layer violation of having MemoizedIdRangeAllocator know about these constructs, but that's a very mild vioaltion, so I agree it's not worth all the casting.